jib-core/src/main/java/com/google/cloud/tools/jib/api/DockerDaemonImage.java

```diff
@@ -55,41 +53,30 @@ public static DockerDaemonImage named(String imageReference)
   }
 
   private final ImageReference imageReference;
-  private Path dockerExecutable = Paths.get("docker");
+
+  private DockerClient dockerClient = DockerClient.newClient();
 
   /** Instantiate with {@link #named}. */
   private DockerDaemonImage(ImageReference imageReference) {
     this.imageReference = imageReference;
   }
 
-  /**
-   * Sets the path to the {@code docker} CLI. This is {@code docker} by default.
-   *
-   * @param dockerExecutable the path to the {@code docker} CLI
-   * @return this
-   */
-  public DockerDaemonImage setDockerExecutable(Path dockerExecutable) {
-    this.dockerExecutable = dockerExecutable;
-    return this;
-  }
-
   @Override
   public ImageConfiguration toImageConfiguration() {
     return ImageConfiguration.builder(imageReference).build();
   }
 
   @Override
   public BuildSteps toBuildSteps(BuildConfiguration buildConfiguration) {
-    return BuildSteps.forBuildToDockerDaemon(
-        DockerClient.newClient(dockerExecutable), buildConfiguration);
+    return BuildSteps.forBuildToDockerDaemon(dockerClient, buildConfiguration);
   }
 
-  /**
-   * Gets the path to the {@code docker} CLI.
-   *
-   * @return the path to the {@code docker} CLI
-   */
-  Path getDockerExecutable() {
-    return dockerExecutable;
+  public DockerClient getDockerClient() {
```

Comment:
Looks like this exists only for testing. It'd be better to avoid this, but if you think you can't, at least add `@VisibleForTesting` for indication (even though it is already public).

Reply:
For now, I left it with @ VisibleForTesting